modules/rostests/apitests/user32/LoadImageGCC.c

```diff
@@ -0,0 +1,238 @@
+            bmi.bmiHeader.biBitCount    = bitmap.bmBitsPixel;
+            bmi.bmiHeader.biCompression = BI_RGB;
+            bmi.bmiHeader.biSizeImage   = 0;
+
+            size = ((bitmap.bmWidth * bmi.bmiHeader.biBitCount + 31) / 32) * 4 * bitmap.bmHeight;
+
+            hMem = GlobalAlloc(GMEM_MOVEABLE, size);
+            lpBits = GlobalLock(hMem);
+            GetDIBits(hdc, hBmp, 0, bitmap.bmHeight, lpBits, &bmi, DIB_RGB_COLORS);
+
+            /* Get one line above bottom line of bitmap */
+            memcpy(img, (VOID *)((INT_PTR)lpBits + 4 * bitmap.bmWidth), 8);
+
+            ok(img[0] == 0, "Byte 0 Bad\n");
+            ok(img[1] == 0, "Byte 1 Bad\n");
+            ok(img[2] == 0, "Byte 2 Bad\n");
+            ok(img[3] == 0, "Byte 3 Bad\n");
+
+            GlobalUnlock(hMem);
+            GlobalFree(hMem);
+
+            DeleteDC(hdcMem);
+
+            EndPaint(hWnd, &ps);
+            break;
+        }
+
+        case WM_DESTROY:
+            PostQuitMessage(0);
+            break;
+        default:
+            return DefWindowProc(hWnd, message, wParam, lParam);
+    }
+    return 0;
+}
+
+static ATOM
+MyRegisterClass(HINSTANCE hInstance)
+{
+    WNDCLASSEXW wcex;
+
+    wcex.cbSize = sizeof(WNDCLASSEX);
+
+    wcex.style         = CS_HREDRAW | CS_VREDRAW;
+    wcex.lpfnWndProc   = WndProc;
+    wcex.cbClsExtra    = 0;
+    wcex.cbWndExtra    = 0;
+    wcex.hInstance     = hInstance;
+    wcex.hIcon         = NULL;
+    wcex.hCursor       = LoadCursor(NULL, IDC_ARROW);
+    wcex.hbrBackground = (HBRUSH)(COLOR_WINDOW+1);
+    wcex.lpszMenuName  = NULL;
+    wcex.lpszClassName = szWindowClass;
+    wcex.hIconSm       = NULL;
+
+    return RegisterClassExW(&wcex);
+}
+
+
+static BOOL
+InitInstance(HINSTANCE hInstance, int nCmdShow)
+{
+   HWND hWnd;
+
+   hWnd = CreateWindowExW(0,
+                          szWindowClass,
+                          L"Bmp test",
+                          WS_OVERLAPPEDWINDOW,
+                          CW_USEDEFAULT,
+                          CW_USEDEFAULT,
+                          300,
+                          120,
+                          NULL,
+                          NULL,
+                          hInstance,
+                          NULL);
+
+   if (!hWnd)
+      return FALSE;
+
+   ShowWindow(hWnd, nCmdShow);
+   UpdateWindow(hWnd);
+
+   return TRUE;
+}
+
+START_TEST(LoadImageGCC)
+{
+    UINT i;
+    WCHAR PathBuffer[MAX_PATH];
+
+    /* Windows 2003 cannot run this test. Testman shows CRASH, so skip it. */
+    if (!IsReactOS())
+        return;
+
+    /* Extract Data Files */
+    for (i = 0; i < _countof(DataFiles); ++i)
+    {
+        ExpandEnvironmentStringsW(DataFiles[i].FileName, PathBuffer, _countof(PathBuffer));
+
+        if (!ResourceToFileW(DataFiles[i].ResourceId, PathBuffer))
+        {
+            printf("ResourceToFile Failed. Exiting now\n");
+            goto Cleanup;
+        }
+    }
+
+    MyRegisterClass(NULL);
```

Comment:
Why is there a window in this test?
I don't think that is needed to load a bitmap?

A memory DC can be created without a window, and this should speed the test up significantly, and make it simpler as well.

Reply:
Good idea. Give me a bit to see what I can do. Thanks.

Reply:
Revised to remove window creation. Thanks.